gap/projective/almostsimple.gi

```diff
@@ -1429,3 +1429,126 @@ FindHomMethodsProjective.SporadicsByOrders := function(ri,G)
   od;
   return false; # FIXME: false = NeverApplicable here really correct?
 end;
+
+#! @BeginChunk NameSporadic
+#! This method returns a list of sporadic simple groups that <A>ri</A>
+#! could be. It does not recognise the monster and the baby monster group. It
+#! is based on the magma v2.24.10 function <C>RecognizeSporadic</C>.
+FindHomMethodsProjective.NameSporadic := function(ri)
+    local orders, setOfOrders, maximalOrders, counter, namesOfPossibleSporadics, i, j;
+    orders := [];
+    #do we need SetPseudoRandomStamp?
+    for i in [1 .. 500] do
+        Add(orders, RandomElmOrd(ri, "NameSporadic", false).order);
+    od;
+    # Compute maximal orders. Maximal in the sense that it does not divide the
+    # order of another group element.
+    setOfOrders := Immutable(Set(orders));
+    maximalOrders := [];
+    for i in setOfOrders do
+        counter := 0;
+        for j in setOfOrders do
+            if j mod i = 0 then
+                counter := counter + 1;
+            fi;
+        od;
+        if counter = 1 then
+            Add(maximalOrders, i);
+        fi;
+    od;
+    namesOfPossibleSporadics := [];
+    if maximalOrders = [5, 6, 8, 11] then
```

Comment:
OK I am getting into micro optimizations here, but let me point out that this code will create a new list `[5,6,8,11]` each time it is executed. So if we call this function a 1000 times, we create 1000 such lists, which all need to be garbage collected later on. We could refactor the case perhaps by adding a global structure like this:
```gap
BindGlobal("SporadicsData", MakeImmutable([
  # entries:  [ name,  superset, subset ]
  [ "M11", [5, 6, 8, 11], [5, 6, 8, 11] ],
  ....
]));
And then the code can look like this:
```gap
  for sd in SporadicsData do
    if IsSubset(sd[2], maximalOrders) and IsSubset(maximalOrders, sd[3]) then
        Add(namesOfPossibleSporadics, sd[1]);
     fi;
  od;
```
Or perhaps `SporadicsData` should be a list of *records*, that might be easier to debug and gives more flexibility
```gap
BindGlobal("SporadicsData", MakeImmutable([
  rec( name:="M11", orders := [5, 6, 8, 11] ),
  ....
  rec( name:="M23", orders := [8, 10, 11, 12, 14, 15, 21, 23], required := [11, 21, 23] ),
  ....
]));
...
  for sd in SporadicsData do
    if not IsSubset(sd.orders, maximalOrders) then continue; fi;
    if IsBound(sd.required) then
      if not IsSubset(maximalOrders, sd.required) then continue; fi;
   else
    if Length(maximalOrders) <> Lenght(sd.orders) then continue; fi;
   fi;
   Add(namesOfPossibleSporadics, sd[1]);
  od;
```

Reply:
I'll try the record approach.